tests/test_text/test_time_intervals.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+import os
+
+from numpy.testing import assert_array_equal
+import pandas as pd
+from pynwb import NWBHDF5IO
+
+from neuroconv.datainterfaces import ExcelTimeIntervalsInterface, CsvTimeIntervalsInterface
+from neuroconv.tools.text import convert_df_to_time_intervals
+
+
+trials_xls_path = os.path.join(os.path.dirname(__file__), "trials.xlsx")
+trials_csv_path = os.path.join(os.path.dirname(__file__), "trials.csv")
+
+
+def test_convert_df_to_time_intervals():
+    df = pd.read_excel(trials_xls_path)
+    time_intervals = convert_df_to_time_intervals(df)
+
+
+def test_convert_df_to_time_intervals_colname_mapping():
+    df = pd.read_excel(trials_xls_path)
+    time_intervals = convert_df_to_time_intervals(df, column_name_mapping=dict(condition="cond"))
+    assert time_intervals.colnames == ("start_time", "stop_time", "cond")
+
+
+def test_convert_df_to_time_intervals_():
+    df = pd.read_excel(trials_xls_path)
+    time_intervals = convert_df_to_time_intervals(
+        df, column_descriptions=dict(condition="This is a custom description")
+    )
+    assert time_intervals["condition"].description == "This is a custom description"
+
+
+def test_excel_time_intervals():
+    interface = ExcelTimeIntervalsInterface(trials_xls_path)
+    metadata = interface.get_metadata()
+    metadata["NWBFile"] = dict(session_start_time=datetime.now().astimezone())
+    nwbfile = interface.run_conversion(metadata=metadata)
+    assert nwbfile.intervals["trials"].colnames == ("start_time", "stop_time", "condition")
+
+
+def test_excel_column_name_mapping():
+    interface = ExcelTimeIntervalsInterface(trials_xls_path)
+    metadata = interface.get_metadata()
+    metadata["NWBFile"] = dict(session_start_time=datetime.now().astimezone())
+    nwbfile = interface.run_conversion(metadata=metadata, column_name_mapping=dict(condition="cond"))
```

Comment:
Can you add a test that we can reference for people demonstrating a source file that does not have the `start_time`/`stop_time` fields and instead relies on the `column_name_mapping` to satisfy those requirements? I think that will actually be quite common

Reply:
sure